homeassistant/components/roborock/select.py

```diff
@@ -2,31 +2,32 @@
 from collections.abc import Callable
 from dataclasses import dataclass
 
-from roborock.code_mappings import RoborockMopIntensityCode, RoborockMopModeCode
 from roborock.containers import Status
-from roborock.exceptions import RoborockException
-from roborock.typing import RoborockCommand
+from roborock.roborock_typing import RoborockCommand
 
 from homeassistant.components.select import SelectEntity, SelectEntityDescription
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import HomeAssistantError
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.util import slugify
 
 from .const import DOMAIN
 from .coordinator import RoborockDataUpdateCoordinator
 from .device import RoborockCoordinatedEntity
-from .models import RoborockHassDeviceInfo
 
 
 @dataclass
 class RoborockSelectDescriptionMixin:
     """Define an entity description mixin for select entities."""
 
+    # The command that the select entity will send to the api.
     api_command: RoborockCommand
+    # Gets the current value of the select entity.
     value_fn: Callable[[Status], str]
-    options_lambda: Callable[[str], list[int]]
+    # Gets all options of the select entity.
+    options_lambda: Callable[[Status], list[str]]
+    # Takes the value from the select entiy and converts it for the api.
+    parameter_lambda: Callable[[tuple[str, Status]], list[int]]
```

Comment:
How about making this accept two parameters rather than a tuple as the arugments: 
```suggestion
    parameter_lambda: Callable[[str, Status], list[int]]
```
Then below it can be:
```
    parameter_lambda=lambda key, status: [
```

Reply:
Optional: One other thought is that you could consider these two functions as a lambda that converts from `Status` to `RoborockEnum`, then let the entity do the rest of the work of calling `.keys` and `.items` to do the work needed.